tt_metal/tt_metal.cpp

```diff
@@ -1139,14 +1154,37 @@ std::unique_ptr<GlobalSemaphore> CreateGlobalSemaphore(
 
 std::shared_ptr<Buffer> CreateBuffer(const InterleavedBufferConfig &config) {
     return Buffer::create(
-        config.device, config.size, config.page_size, config.buffer_type, config.buffer_layout, std::nullopt, std::nullopt);
+        config.device,
+        config.size,
+        config.page_size,
+        config.buffer_type,
+        config.buffer_layout,
+        std::nullopt,
+        std::nullopt,
+        std::nullopt);
```

Comment:
this can probably be cleaned up in a separate PR but could we add these extra params to the config? I think some are common between Interleaved and Sharded configs but having a bunch of nullopts and then sub device ids looks awkward. Could you create an issue for this

Reply:
Yes, I'll file an issue for this. The three nullopts are shard_parameters, bottom_up (this can probably go into the config), and sub_device_id. Do we want sub_device_id in the config as well, or does it make more sense as it is right now where host_api takes it as a separate argument like device addr?

Reply:
since it isn't really parameter of how to set up the buffer I think its fine for it to go outside the config